clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -230,22 +231,36 @@ private boolean canSendRequest(String node) {
 
     /**
      * Queue up the given request for sending. Requests can only be sent out to ready nodes.
-     *
      * @param request The request
      * @param now The current timestamp
      */
     @Override
     public void send(ClientRequest request, long now) {
-        String nodeId = request.request().destination();
+        doSend(request, false, now);
+    }
+
+    private void sendInternalMetadataRequest(ClientRequest request, long now) {
+        doSend(request, true, now);
```

Comment:
Good idea. I think we can refactor this so that the argument itself is an instance of `MetadataRequest`.